notes/Sources/Service/BaseAPI.swift

```diff
@@ -12,6 +13,14 @@ class BaseAPI<T: TargetType> {
     self.provider = provider
     self.callbackQueue = callbackQueue
   }
+  
+  func future<Response: Decodable>(_ target: T) -> Future<Response, Error> {
```

Comment:
- Why don't you use default **Moya's**  extension for **Combine**?
- This is not extension of BaseAPI (and even if it was, it would have called **request**, not **future**...)
- consider dropping it and use provided API